source/common/stats/isolated_store_impl.h

```diff
@@ -268,9 +243,126 @@ class IsolatedStoreImpl : public StoreImpl {
   IsolatedStatsCache<TextReadout> text_readouts_;
   RefcountPtr<NullCounterImpl> null_counter_;
   RefcountPtr<NullGaugeImpl> null_gauge_;
-  ScopeSharedPtr default_scope_;
+  mutable ScopeSharedPtr lazy_default_scope_;
```

Comment:
I could not find a reason that requires this to be mutable. Maybe it needed to be before but not anymore? (Or I missed it)

Reply:
Quite right! It is conceptually mutable, because there's a constRootScope() method which can write it. That is implemented with a const_cast to avoid duplicating code, but conceptually I think lazy_default_scope_ should be considered mutable.

If we didn't use a const_cast in the impl it would have to be mutable (as well as the symbol table).